### PR TITLE
docs: update docs for MTKv9

### LINKED ISSUE #1968: Overridden default set on observed quantity not propagating to its alias
```julia
using ModelingToolkit, Plots, DifferentialEquations

@variables t

@connector function Pin(;name)
    sts = @variables v(t)=1.0 i(t)=1.0 [connect = Flow]
    ODESystem(Equation[], t, sts, []; name=name)
end

function Ground(;name)
    @named g = Pin()
    eqs = [g.v ~ 0]
    compose(ODESystem(eqs, t, [], []; name=name), g)
end

function OnePort(;name)
    @named p = Pin()
    @named n = Pin()
    sts = @variables v(t)=1.0 i(t)=1.0
    eqs = [
           v ~ p.v - n.v
           0 ~ p.i + n.i
           i ~ p.i
          ]
    compose(ODESystem(eqs, t, sts, []; name=name), p, n)
end

function Resistor(;name, R = 1.0)
    @named oneport = OnePort()
    @unpack v, i = oneport
    ps = @parameters R=R
    eqs = [
           v ~ i * R
          ]
    extend(ODESystem(eqs, t, [], ps; name=name), oneport)
end

function Capacitor(;name, C = 1.0)
    @named oneport = OnePort()
    @unpack v, i = oneport
    ps = @parameters C=C
    D = Differential(t)
    eqs = [
           D(v) ~ i / C
          ]
    extend(ODESystem(eqs, t, [], ps; name=name), oneport)
end

function ConstantVoltage(;name, V = 1.0)
    @named oneport = OnePort()
    @unpack v = oneport
    ps = @parameters V=V
    eqs = [
           V ~ v
          ]
    extend(ODESystem(eqs, t, [], ps; name=name), oneport)
end

function Inductor(;name, L = 1.0)
    @named oneport = OnePort()
    @unpack v, i = oneport
    pars = @parameters L = L
    D = Differential(t)

    eqs = [

       #D(v) ~ i / C
       D(i) ~ 1/L * v,
    ]
    extend(ODESystem(eqs, t, [], pars; name = name), oneport)
end


R = 1.0
C = 1.0
V = 20
L = 1.0


@named resistor   =  Resistor(R=R)
@named capacitor  =  Capacitor(C=C)
@named source     =  ConstantVoltage(V=V)
@named inductor   =  Inductor(L=L)
@named ground     =  Ground()

rc_eqs = [
    connect(source.p, inductor.p)
    connect(source.n, inductor.n)
    #connect(capacitor.n, source.n)
    connect(source.n, ground.g)
         ]

@named _rc_model = ODESystem(rc_eqs, t)
@named rc_model = compose(_rc_model,
                          [inductor, source, ground])
sys = structural_simplify(rc_model)
u0 = [
      inductor.v => -1
     ]
prob = ODAEProblem(sys, u0, (0, 10.0))
sol = solve(prob, Tsit5())
plot(sol)
plot(sol(0:0.1:10, idxs = inductor.i),label='A')
#plot(sol(inductor.v),label='A')
sol(0:0.1:10, idxs = inductor.v)
```

To put it simply, I connect the inductor and the 20V voltage source together, and then set the initial value of the inductor voltage to -1. It stands to reason that this situation does not exist and the program should report an error. Why does MTK still default the initial value of the voltage across the inductor to 20? Is this a bug of mtk?

### LINKED ISSUE #2059: `substitute(::ODESystem)` does not substitute in events
Should there be a warning that this only supports substituting within the equations? For example, it doesn’t handle events?

_Originally posted by @isaacsas in https://github.com/SciML/ModelingToolkit.jl/issues/2054#issuecomment-1406665796_

### LINKED ISSUE #2481: Pretty printing list of parameters.
In MTK, I would like to obtain a printable list of parameters along with all their "descriptions": default value, description field, and any other metdata field. Just parameters(prob/sys) does not do this, it only prints the parameters in latex notation but without any metadata. Is it straightforward to get the nice string with metadata that is printed when the whole system itself is printed?

### PR DESCRIPTION
**Pull request by AayushSabharwal**
Close #2059
Close #1968
Close #2481 
 
## Checklist

- [ ] Appropriate tests were added
- [ ] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API

### CODE REVIEW

File: src/variables.jl [hunk 1]
```diff
@@ -15,6 +15,57 @@ Symbolics.option_to_metadata_type(::Val{:irreducible}) = VariableIrreducible
 Symbolics.option_to_metadata_type(::Val{:state_priority}) = VariableStatePriority
 Symbolics.option_to_metadata_type(::Val{:misc}) = VariableMisc
 
+function dump_variable_metadata(var)
```
Comment (Datseris) [inline src/variables.jl:18]: `dump_variable_metadata(var)` seems perfect to me. The named tuple return type allows direct integration with Tables.jl.

Comment (AayushSabharwal) [reply src/variables.jl:18]: Note that the returned `NamedTuple` may not have the same type or fields for all variables. It depends on the types of the symbolic variables themselves as well as the contained metadata

File: docs/src/tutorials/stochastic_diffeq.md [hunk 2]
```diff
@@ -38,5 +34,5 @@ parammap = [
 ]
 
 prob = SDEProblem(de, u0map, (0.0, 100.0), parammap)
-sol = solve(prob, SOSRI())
+sol = solve(prob, LambdaEulerHeun())
```
Comment (ChrisRackauckas) [inline docs/src/tutorials/stochastic_diffeq.md:37]: @YingboMa is it not able to detect it's diagonal in the lowering process?

**Comment by AayushSabharwal** [comment]
This should be good to go, forgot to squash the `fixup!`s last push

File: docs/src/tutorials/stochastic_diffeq.md (hunk 2 above)
Comment (AayushSabharwal) [reply docs/src/tutorials/stochastic_diffeq.md:37]: Opening

Comment (AayushSabharwal) [reply docs/src/tutorials/stochastic_diffeq.md:37]: https://github.com/SciML/ModelingToolkit.jl/issues/2490

**Comment by AayushSabharwal** [comment]
`SteadyStateDiffEq` and `DiffEqCallbacks` have some compat issues with each other

File: src/variables.jl (hunk 1 above)
Comment (Datseris) [reply src/variables.jl:30]: Oh yeah, true. I don't think this is compatible with Tables.jl, but I definitely prefer this option that does not print unecessary informatioN!

**Comment by ChrisRackauckas** [comment]
> Docs still won't build because of structural_simplify not working for OptimizationSystems

Disable that for now, it's fixed in https://github.com/SciML/ModelingToolkit.jl/pull/2403 but we shouldn't wait

**Comment by AayushSabharwal** [comment]
> Disable that for now, it's fixed in https://github.com/SciML/ModelingToolkit.jl/pull/2403 but we shouldn't wait

Done

**Review by ChrisRackauckas — APPROVED** [review]

**Pull request merged**